cylc/flow/cfgspec/suite.py

```diff
@@ -1376,6 +1368,7 @@ def upg(cfg, descr):
         ['cylc', 'health check interval'])
     u.obsolete('8.0.0', ['runtime', '__MANY__', 'job', 'shell'])
     u.obsolete('8.0.0', ['runtime', '__MANY__', 'events', 'mail retry delays'])
+    u.obsolete('8.0.0', ['runtime', '__MANY__', 'events', 'mail smtp'])
```

Comment:
It's just occurred to me, should this be deprecated instead? 
```suggestion
    u.deprecate(
        '8.0.0',
        ['runtime', '__MANY__', 'events', 'mail smtp'],
        ['cylc', 'mail', 'smtp'])
```

Reply:
I don't think so, however, it might make some sense to grab the `root` value:

```
['runtime', 'root', 'events', 'mail smtp']
```

Reply:
Oh yeah, otherwise they would all clash with each other.

There doesn't seem to be a straightforward way to upgrade the `root` one while logging the obsolescent `__MANY__` ones. How about just this instead
```suggestion
    u.deprecate(
        '8.0.0',
        ['runtime', '__MANY__', 'events', 'mail smtp'],
        None,  # This is really a .obsolete(), just with a custom message
        cvtr=converter(lambda x: x,
                       'DELETED (OBSOLETE) - use "[cylc][mail]smtp" instead'))
```